base/server/python/pki/server/__init__.py

```diff
@@ -1097,7 +1097,12 @@ def load_subsystems(self):
         for subsystem_name in SUBSYSTEM_TYPES:
 
             subsystem_dir = os.path.join(self.base_dir, subsystem_name)
-            if not os.path.exists(subsystem_dir):
+            try:
+                if not os.listdir(subsystem_dir):
+                    # Directory exists but it is empty
+                    continue
+            except FileNotFoundError:
+                # Directory does not exist
```

Comment:
Is there a reason why we can't simply create the directory if it doesn't exist? So:
```
try:
    os.makedirs(subsystem_dir)
except FileExistsError:
    pass
```

Reply:
From what I can tell, currently (before my change) this method scans the `base_dir` for folders that correspond with the subsystem names `['ca', 'kra', 'ocsp', 'tks', 'tps']`.

If it finds a folder in that list it proceeds and assumes that subsystem is installed and tries to start it.

I don't think we want to create empty directories as it would make the proceeding code try to start all the subsystems whether they are installed or not

Reply:
OK that makes sense, cheers!

Reply:
Yeah, the code is supposed to create the subsystem object only if the subsystem actually exists (i.e. created by `pkispawn`). The old code assumes that if `/var/lib/pki/pki-tomcat/<subsystem>` exists then the subsystem must have been created, but that turns out to be insufficient since the folder could exist due to incomplete cleanup or as a mounting point to load an existing subsystem configuration from another file system.

One thing I'd suggest is to keep the code flat whenever possible to make it easier to read, so in this case use the regular if-statements instead of try-except clause:
```
if not os.path.exists(subsystem_dir):
    # Directory does not exist
    continue

if not os.listdir(subsystem_dir):
    # Directory exists but it is empty
    continue
```
Feel free to update if you wish. I'll ACK the PR regardless. Thanks!

Reply:
Updated with your suggestion @edewata 

It is much more readable now. Is merging into `master` correct for this change?

Reply:
Thanks! Yes, we should always fix the master branch if applicable, then we cherry pick to a branch (e.g. v10.11) if it's required by a platform that uses that branch (e.g. RHEL 8.x).